### DISCUSSION TITLE
How to generate image with clear background on faceswap mode.

### DISCUSSION THREAD

**Discussion by oldhand7** [lllyasviel/Fooocus#2980] [Q&A] [answered]
I tried to generate images with text prompt and image prompt. I set the faceswap mode.
The others are default.

I am curious why the generated images are blurry background not clear one.
Please help me generate clear background, better backgorund quality photo.

Thanks.

**Comment by IPv6** [comment, marked as answer]
Afaik it`s not really possible to force background to be "clear" or something, there always some gradient.
Better to use separate tools for clearing background - there is a lot of "remove background" tools out there

**Reply by mashb1t** [reply]
You can also use my fork of Fooocus where rembg is already included but you might have to tweak one line of code.
https://github.com/mashb1t/Fooocus/blob/dd5264dd14fdeedec57c87567e008b10a8fcda69/extras/inpaint_mask.py#L40
Change this to False and you get the image without background and you can even prompt which things to select.

**Reply by IPv6** [reply]
Yep, quite useful for advanced image modifications!

**Comment by xhoxye** [comment]
This has nothing to do with the faceswap mode. The trick is that you describe the detailed background
![QQ截图20240523061719](https://github.com/lllyasviel/Fooocus/assets/129571231/1eb0c151-41ab-4044-ab93-78a537c2cd41)
![QQ截图20240523062302](https://github.com/lllyasviel/Fooocus/assets/129571231/3941f7f3-cb91-49fc-a7f3-fbefa9d637b3)

**Reply by minecraftsanta** [reply]
Where is the option to input the  "Fooocus V2 Expansion prompt"

**Reply by mashb1t** [reply]
@minecraftsanta Fooocus V2 is an automated prompt expansion based on GPT2, which you can enable/disable by selecting the style Fooocus V2.
There is no input as this is generated.
